lib/combineNodes.js

```diff
@@ -0,0 +1,367 @@
+'use strict';
+var Cesium = require('cesium');
+var jp = require('jsonpath');
+var _ = require('underscore');
```

Comment:
Other than deep-equals, I'm not sure why we would need any other libraries.  .find is just a for loop with a break statement.  It provides no benefit.  We already have binarySearch in Cesium if your arrays are sorted.  If they aren't sorted, then there is no "optimized" search method; it's always a linear search.